src/units.cpp

```diff
@@ -212,6 +212,19 @@ UnitsPtr Units::create(const std::string &name) noexcept
 
 bool Units::isBaseUnit() const
 {
+    if (isImport()) {
+        // If our model is unresolved or our import source is unresolved then return false
+        ImportSourcePtr importedSource = importSource();
+        if (importedSource != nullptr) {
+            ModelPtr model = importedSource->model();
+            if (model != nullptr && model->hasUnits(importReference())) {
+                auto unit = model->units(importReference());
+                return unit->unitCount() == 0;
```

Comment:
I think here we should be doing `return unit->isBaseUnit();` to handle the case when you're importing an imported units...if I have my types correct ;) Then we only have the check on `unitCount()` defined in one place and handle an arbitrarily deep import hierarchy.